File: Basic_Math/armstrong_number/armstrongNumber.go
package main

import (
	"fmt"
	"math"
	"strconv"
)

func armstrongNumber(n int) bool {
	digitCount := len(strconv.Itoa(n))
	temp := n
	var result float64
	for temp > 0 {
		lastDigit := temp % 10
		result += math.Pow(float64(lastDigit), float64(digitCount))
		temp /= 10
	}
	return int(result) == n
}

func main() {
	fmt.Println("Armstrong number:", armstrongNumber(1634))
}

File: Basic_Math/armstrong_number/armstrongNumber.py
def armstrongNumber(n):
    digitCount = len(str(n))
    temp = n
    result = 0;
    while ( temp > 0 ):
        lastDigit = temp % 10
        result += lastDigit ** digitCount
        temp = temp // 10
    return result == n

print("armstrong number :",armstrongNumber(1634))

